### PR TITLE
[MAJOR] Implemented frontend for call logs.

### LINKED ISSUE #153: Implement call logs frontend page to reflect UI mockups
Epic/User story #130

### PR DESCRIPTION
**Pull request by jmezaber**
closes #153 

UI
Screen Shot 2022-01-15 at 5 39 27 PM

New call log pop-up
Screen Shot 2022-01-15 at 5 39 43 PM

Demo of edit and delete log
edit_and_delete_log_demo

### CODE REVIEW

File: frontend/src/dto/CallLogs/CallLogDTOs.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  callerEmail?: string;
+}
+
+export interface CallCreationDTO {
+  receiverName?: string;
+  date: string;
+  phoneNumber?: string;
+  description: string;
+  receiverEmail: string;
+  action: string;
```
Comment (huangs08) [inline frontend/src/dto/CallLogs/CallLogDTOs.ts:19]: @nasaku898 Same comment as previous PR (action should be of type ENUM). Should we make a ticket to handle this?

File: frontend/src/components/CommunicationLogs/ViewCallLogs.tsx [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+      receiverEmail: selectedClient as string,
+      description: addedNote as string,
+      action: actions[selectedIndexAction].toUpperCase(),
+      date: new Date().toISOString().split('T')[0],
+      followUp: false,
+      neverCallBack: false,
+      callerEmail: account.account.email,
+    };
+    try {
+      console.log(data);
```
Comment (nasaku898) [inline frontend/src/components/CommunicationLogs/ViewCallLogs.tsx:169]: remove console.log

File: frontend/src/components/CommunicationLogs/ViewCallLogs.tsx [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+      };
+      display.push(dataDisplay);
+    });
+    setCallsDisplayed(display);
+  };
+
+  useEffect(() => {
+    fetchData();
+    // eslint-disable-next-line
+  }, [select]);
```
Comment (nasaku898) [inline frontend/src/components/CommunicationLogs/ViewCallLogs.tsx:236]: try adding fetchData to the array and remove the comment. It should work.

**Review by nasaku898 — CHANGES_REQUESTED** [review]
Overall good, added some few comments.

File: frontend/src/components/CommunicationLogs/ViewCallLogs.tsx (hunk 2 above)
Comment (jmezaber) [reply frontend/src/components/CommunicationLogs/ViewCallLogs.tsx:169]: good catch!

File: frontend/src/components/CommunicationLogs/ViewCallLogs.tsx (hunk 3 above)
Comment (bowsong) [reply frontend/src/components/CommunicationLogs/ViewCallLogs.tsx:235]: Wouldn't that create an infinite loop as every time fetchData is called, useEffect might also be called.

Comment (jmezaber) [reply frontend/src/components/CommunicationLogs/ViewCallLogs.tsx:235]: The way it is right now, it doesn't create an infinite loop

File: frontend/src/dto/CallLogs/CallLogDTOs.ts (hunk 1 above)
Comment (nasaku898) [reply frontend/src/dto/CallLogs/CallLogDTOs.ts:19]: Yeah I think we can create a seperate ticket just so we can merge this one.

**Review by nasaku898 — APPROVED** [review]
Great work

**Review by huangs08 — APPROVED** [review]

**Pull request merged**